File: DoublePointer/11ContainerWithMostWater.cpp
/**
 Solution:
 Two pointers
*/
class Solution {
public:
    int maxArea(vector<int>& height) {
        int rev = 0;
        int i = 0, j = height.size()-1;
        while(j>i){
            rev = max(rev, min(height[i],height[j])*(j-i));
            if(height[i]<height[j]) i++;
            else j--;
        }
        return rev;
    }
};

File: DoublePointer/11containerWithMostWater.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        def check(i,j):
            return (j-i)*min(height[i],height[j])
        
        res = 0
        i, j = 0, len(height)-1
        while i < j:
            res = max(res,check(i,j))
            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
        return res